request.go

```diff
@@ -143,6 +143,7 @@ func prepareRequest(req *transport.Request, headers map[string]string, opts Opti
 
 	// Add request metadata
 	req.TargetService = opts.TOpts.ServiceName
+	req.ShardKey = opts.TOpts.ShardKey
```

Comment:
Is this a bugfix or required for gRPC? Maybe this should be a separate change with a separate test?

Reply:
I guess a little of both? I think a separate PR might be a little overkill, but there's a lot of conflation between the request and the options given to each transport implementation. This is just adding another set for request.

I can write a test for this.

Reply:
Testing added.